e2e/nomostest/config_sync.go

```diff
@@ -816,6 +817,128 @@ func RepoSyncObjectV1Beta1(nn types.NamespacedName, repoURL string, sourceFormat
+func (nt *NT) UpdateRootSyncSpec(rs *v1beta1.RootSync) error {
+	var clusterRootSync v1beta1.RootSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRootSync); err != nil {
+		return err
+	}
+	clusterRootSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRootSync)
+}
+
+// UpdateRepoSyncSpec updates the RepoSync object on the cluster with the provided
+// spec.
+func (nt *NT) UpdateRepoSyncSpec(rs *v1beta1.RepoSync) error {
+	var clusterRepoSync v1beta1.RepoSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRepoSync); err != nil {
+		return err
+	}
+	clusterRepoSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRepoSync)
+}
+
+// RootSyncObjectOCI returns a RootSync object that syncs the provided OCIImage.
+func (nt *NT) RootSyncObjectOCI(name string, image *registryproviders.OCIImage) *v1beta1.RootSync {
+	rs := RootSyncObjectV1Beta1FromRootRepo(nt, name)
+	rs.Spec.SourceType = string(v1beta1.OciSource)
+	rs.Spec.Oci = &v1beta1.Oci{
+		Image: image.FloatingBranchTag(),
+		Auth:  configsync.AuthNone,
+	}
+	switch *e2e.OCIProvider {
+	case e2e.Local:
+		// Local provider requires CA cert because host is not well known
+		rs.Spec.Oci.CACertSecretRef = &v1beta1.SecretReference{
+			Name: PublicCertSecretName(RegistrySyncSource),
+		}
+	case e2e.ArtifactRegistry:
+		// AR provider requires auth because the registry is private
+		rs.Spec.Oci.Auth = configsync.AuthGCPServiceAccount
+		rs.Spec.Oci.GCPServiceAccountEmail = registryproviders.ArtifactRegistryReaderEmail()
+	}
+	return rs
+}
+
+// RepoSyncObjectOCI returns a RepoSync object that syncs the provided OCIImage.
+func (nt *NT) RepoSyncObjectOCI(nn types.NamespacedName, image *registryproviders.OCIImage) *v1beta1.RepoSync {
+	rs := RepoSyncObjectV1Beta1FromNonRootRepo(nt, nn)
+	rs.Spec.SourceType = string(v1beta1.OciSource)
+	rs.Spec.Oci = &v1beta1.Oci{
+		Image: image.FloatingBranchTag(),
+		Auth:  configsync.AuthNone,
+	}
+	switch *e2e.OCIProvider {
+	case e2e.Local:
+		// Local provider requires CA cert because host is not well known
+		rs.Spec.Oci.CACertSecretRef = &v1beta1.SecretReference{
+			Name: PublicCertSecretName(RegistrySyncSource),
+		}
+	case e2e.ArtifactRegistry:
+		// AR provider requires auth because the registry is private
+		rs.Spec.Oci.Auth = configsync.AuthGCPServiceAccount
+		rs.Spec.Oci.GCPServiceAccountEmail = registryproviders.ArtifactRegistryReaderEmail()
+	}
+	return rs
+}
+
+// RootSyncObjectHelm returns a RootSync object that syncs the provided HelmPackage
+func (nt *NT) RootSyncObjectHelm(name string, chart *registryproviders.HelmPackage) *v1beta1.RootSync {
+	rs := RootSyncObjectV1Beta1FromRootRepo(nt, name)
+	rs.Spec.SourceType = string(v1beta1.HelmSource)
+	rs.Spec.Helm = &v1beta1.HelmRootSync{
+		HelmBase: v1beta1.HelmBase{
+			Repo:    nt.HelmProvider.SyncURL(chart.Name),
+			Chart:   chart.Name,
+			Version: chart.Version,
+			Auth:    configsync.AuthNone,
+			Period:  metav1.Duration{Duration: 15 * time.Second},
```

Comment:
Maybe move this timeout to a const with a comment as to why we're shortening it and not testing the normal behavior.

Are we testing behavior that normally would take an hour to complete? Is there some way we can do it that matches user behavior better? Or is this the case where we need to make the behavior more complicated and this is just a temporary workaround? If so, maybe add a TODO on the const.

Reply:
done